commons/src/connections/availability.ts

```diff
@@ -63,3 +66,82 @@ export const fetchAvailability = async (
     })
     .catch((error) => handleError(query.component_id, error));
 };
+
+export const fetchConsecutiveAvailability = async (
+  query: ConsecutiveAvailabilityQuery,
+): Promise<ConsecutiveEvent[][]> => {
+  return fetch(
+    `${getMiddlewareApiUrl(
+      query.component_id,
+    )}/calendars/availability/consecutive`,
+    getFetchConfig({
+      method: "POST",
+      component_id: query.component_id,
+      access_token: query.access_token,
+      body: query.body,
+    }),
+  )
+    .then(async (apiResponse): Promise<ConsecutiveEvent[][]> => {
+      const json = await handleResponse<
+        MiddlewareResponse<PreDatedTimeSlot[][]>
+      >(apiResponse);
+      let response: PreDatedTimeSlot[][] =
+        json.response?.map((blockSlot) => {
+          blockSlot = blockSlot.map((slot: any) => {
+            slot.start_time = new Date(slot.start_time * 1000);
+            slot.end_time = new Date(slot.end_time * 1000);
+            return slot;
+          });
+          return blockSlot;
+        }) || [];
+      const hydratedResponse = hydrateSlotsToEvents(
+        response,
+        query.body.events,
+      );
+      const dedupedResponse =
+        removeSimultaneousAvailabilityOptions(hydratedResponse);
+      return dedupedResponse;
+    })
+    .catch((error) => handleError(query.component_id, error));
+};
+
+// Doing the best we can with what we've got: /calendars/availability/consecutive doesn't return any info other than emails
+// and start/end times. This means that if we have to events (EventDefinitions) with the same email addresses? We're shooting in the dark about which is which.
+// TODO: allow for an indicator on the API side
+function hydrateSlotsToEvents(
+  availabilities: PreDatedTimeSlot[][],
+  events: EventDefinition[],
+): ConsecutiveEvent[][] {
+  return availabilities.map((block) => {
+    return block.map((subevent) => {
+      return {
+        ...subevent,
+        ...events.find(
+          (eventdef) =>
+            eventdef.participantEmails.length === subevent.emails.length &&
```

Comment:
@philrenaud I am getting an error here, using the demo data:
```
availability.ts:121 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'length')
    at availability.ts:121
```

Looks like `eventdef.participantEmails` is undefined 🤔

Reply:
It looks like this is happening in the scheduler/index.html only because the events objects passed do not match the type definition of `EventDefinition`. Should `participantEmails` be a required property?